services/core/shared/indexdb/mysql.js

```diff
@@ -318,12 +318,23 @@ const getDbInstance = async (tableName, tableConfig, connEndpoint = config.endpo
 		return resultSet;
 	};
 
-	const increment = async params => knex.transaction(
-		trx => trx(tableName)
-			.where(params.where.property, '=', params.where.value)
-			.increment(params.increment)
-			.transacting(trx),
-	);
+	const increment = async (params, row) => {
+		try {
+			const [result] = await knex.transaction(
+				trx => trx(tableName)
+					.insert(row)
+					.transacting(trx),
+			);
+			return result;
+		} catch (error) {
+			return knex.transaction(
+				trx => trx(tableName)
+					.where(params.where.property, '=', params.where.value)
+					.increment(params.increment)
+					.transacting(trx),
+			);
+		}
```

Comment:
Please consider replacing the `try...catch` with a simpler `if..else` conditional.
Also, could you please confirm if the return values in both the cases are consistent?
```suggestion
		if (row) {
			const [result] = await knex.transaction(
				trx => trx(tableName)
					.insert(row)
					.transacting(trx),
			);
			return result;
		} else {
			return knex.transaction(
				trx => trx(tableName)
					.where(params.where.property, '=', params.where.value)
					.increment(params.increment)
					.transacting(trx),
			);
		}
```

Reply:
It has been clarified offline and response is consistent now.